pkg/resources/custom_diffs.go

```diff
@@ -2,51 +2,74 @@ package resources
-			return client.Parameters.ShowAccountParameter(context.Background(), parameter)
-		},
-		func(v any) string { return strconv.Itoa(v.(int)) },
-	)
+func AccountObjectStringValueComputedIf(key string, params []*sdk.Parameter, parameterLevel sdk.ParameterType, parameter sdk.AccountParameter) schema.CustomizeDiffFunc {
+	return ValueComputedIf(key, params, parameterLevel, parameter, func(value any) string { return value.(string) })
 }
 
-// NestedValueComputedIf internally calls schema.ResourceDiff.SetNewComputed whenever the inner function returns true.
-// It's main purpose was to use it with hierarchical values that are marked with Computed and Optional. Such values should
-// be recomputed whenever the value is not in the configuration and the remote value is not equal to the value in state.
-func NestedValueComputedIf(key string, showParam func(client *sdk.Client) (*sdk.Parameter, error), valueToString func(v any) string) schema.CustomizeDiffFunc {
-	return customdiff.ComputedIf(key, func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+func AccountObjectIntValueComputedIf(key string, params []*sdk.Parameter, parameterLevel sdk.ParameterType, parameter sdk.AccountParameter) schema.CustomizeDiffFunc {
+	return ValueComputedIf(key, params, parameterLevel, parameter, func(value any) string { return strconv.Itoa(value.(int)) })
+}
+
+func AccountObjectBoolValueComputedIf(key string, params []*sdk.Parameter, parameterLevel sdk.ParameterType, parameter sdk.AccountParameter) schema.CustomizeDiffFunc {
+	return ValueComputedIf(key, params, parameterLevel, parameter, func(value any) string { return strconv.FormatBool(value.(bool)) })
+}
+
+func ValueComputedIf(key string, parameters []*sdk.Parameter, objectParameterLevel sdk.ParameterType, accountParameter sdk.AccountParameter, valueToString func(v any) string) schema.CustomizeDiffFunc {
+	var parameterValue *string
+	var parameterLevel *sdk.ParameterType
+
+	for _, parameter := range parameters {
+		if parameter.Key == string(accountParameter) {
+			parameterLevel = &parameter.Level
+			parameterValue = &parameter.Value
+			break
+		}
+	}
+
+	condition := func(ctx context.Context, d *schema.ResourceDiff, meta any) bool {
 		configValue, ok := d.GetRawConfig().AsValueMap()[key]
-		if ok && len(configValue.AsValueSlice()) == 1 {
+
+		if parameterLevel == nil || parameterValue == nil {
+			log.Printf("[ERROR] ValueComputedIf, parameter %s not found", accountParameter)
 			return false
 		}
 
-		client := meta.(*provider.Context).Client
+		// For cases where currently set value (in the config) is equal to the parameter, but not set on the right level.
+		// The parameter is set somewhere higher in the hierarchy, and we need to "forcefully" set the value to
+		// perform the actual set on Snowflake (and set the parameter on the correct level).
+		if *parameterLevel != objectParameterLevel && !configValue.IsNull() && *parameterValue == valueToString(d.Get(key)) {
```

Comment:
*unset it, because the configuration is empty here, and yes you're right that the first check is obsolete (your test passed with the same plans and asserts).